plugins/obs-outputs/rtmp-stream.c

```diff
@@ -922,33 +961,41 @@ static inline bool send_headers(struct rtmp_stream *stream)
 
 	if (!send_audio_header(stream, i++, &next))
 		return false;
-
-	// send metadata only if HDR
-	video_t *video = obs_get_video();
-	const struct video_output_info *info = video_output_get_info(video);
-	enum video_colorspace colorspace = info->colorspace;
-	if (colorspace == VIDEO_CS_2100_PQ || colorspace == VIDEO_CS_2100_HLG)
-		if (!send_video_metadata(stream)) // Y2023 spec
-			return false;
-
-	if (!send_video_header(stream))
+	if (!send_video_metadata(stream, 0) || !send_video_header(stream, 0))
 		return false;
 
 	while (next) {
 		if (!send_audio_header(stream, i++, &next))
 			return false;
 	}
 
+	i = 1;
+	while (obs_output_get_video_encoder2(stream->output, i) != NULL) {
+		if (!send_video_metadata(stream, i) ||
+		    !send_video_header(stream, i))
+			return false;
+		i += 1;
+	}
+
```

Comment:
Is there any situation where there could be a gap in encoders? i.e. 0, 1, 3 are valid but 2 is empty?

Either way I'd personally write this as a for-loop:
```c
for (size_t idx = 1; idx < MAX_OUTPUT_VIDEO_ENCODERS; idx++) {
    obs_encoder_t *enc = obs_output_get_video_encoder2(stream->output, idx);
    if (!enc)
        continue;
    if (!send_video_metadata(stream, idx) || !send_video_header(stream, idx))
        return false;
}
```

Reply:
the beta multitrack video output doesn't create gaps, but there shouldn't be a reason to not support them

Reply:
@derrod: might be interesting to also change this for audio tracks

Reply:
changed in https://github.com/obsproject/obs-studio/pull/10500#event-12641499492